### DISCUSSION TITLE
Lesson 11: IPFS .png link loads, but not .json

### DISCUSSION THREAD

**Discussion by hai-yr** [smartcontractkit/full-blockchain-solidity-course-py#959] [Q&A]
I deployed and created my advanced collectible from lesson 11. The issue I am having is that the .json ipfs page never loads (I always get 504 timeout) while the .png does. This also results in the opensea testnet link never showing the image even when I refresh the metadata.

https://ipfs.io/ipfs/QmYEWjVVXHMEHNZ219a2hoytB6JiQximigkfeeNZBct2TP?filename=windswept.png
https://ipfs.io/ipfs/QmYEFoDo1ztBBXP83afPvovAHcoQjjbV5f3tcE1MQi5jXa?filename=0-WINDSWEPT.json

Here is the opensea testnet link
https://testnets.opensea.io/assets/0xe604Ca19743d8d956a8F64036F1EdE566c28C336/0

**Comment by cromewar** [comment]
Hello @hai-yr interesting, could you share more info on the error, which mechanism did you used to get the json file? because it also gives me 505 error and in inaccessible.

**Reply by hai-yr** [reply]
Here is the contents of the .json file from the metadata_template that is created before attempted upload to ipfs

`{"name": "WINDSWEPT", "description": "redo of one of three ancient bonsais", "image": "https://ipfs.io/ipfs/QmYEWjVVXHMEHNZ219a2hoytB6JiQximigkfeeNZBct2TP?filename=windswept.png", "attributes": [{"trait_type": "moon", "value": ""}]}`

and here is the create_metadata code:

```Python
from brownie import AdvancedCollectible, network
from scripts.helpful_scripts import get_kind
from metadata.metadata import metadata_template
from pathlib import Path
import requests
import json
import os

kind_to_image_uri = {} # I don't need this because i commented out the code that uses it


def main():
    advanced_collectible = AdvancedCollectible[-1]
    number_of_advanced_collectibles = advanced_collectible.tokenCounter()
    for token_id in range(number_of_advanced_collectibles):
        kind = get_kind(advanced_collectible.tokenIdToKind(token_id))
        metadata_file_name = (
            f"./metadata/{network.show_active()}/{token_id}-{kind}.json"
        )
        collectible_metadata = metadata_template
        if Path(metadata_file_name).exists():
            print("file already exists, delete to overwrite")
        else:
            print("creating metadata file")
            collectible_metadata["name"] = kind
            collectible_metadata["description"] = "redo of one of three ancient bonsais"
            # collectible_metadata["image_uri"] = ?
            # image_uri = upload_to_ipfs()
            image_path = "./img/" + kind.lower().replace("_", "-") + ".png"
            image_uri = None
            # if os.getenv("UPLOAD_IPFS") == "true":
            # image_uri = upload_to_ipfs(image_path)
            image_uri = "https://ipfs.io/ipfs/QmYEWjVVXHMEHNZ219a2hoytB6JiQximigkfeeNZBct2TP?filename=windswept.png"  # image_uri if image_uri else kind_to_image_uri[kind]
            collectible_metadata["image"] = image_uri
            with open(metadata_file_name, "w") as file:
                json.dump(collectible_metadata, file)
            if os.getenv("UPLOAD_IPFS") == "true":
                upload_to_ipfs(metadata_file_name)


def upload_to_ipfs(filepath):  # could swap with pinata
    with Path(filepath).open("rb") as fp:
        image_binary = fp.read()
        ipfs_url = "http://127.0.0.1:5001"
        endpoint = "/api/v0/add"
        response = requests.post(ipfs_url + endpoint, files={"file": image_binary})
        ipfs_hash = response.json()["Hash"]
        filename = filepath.split("/")[-1:][0]
        image_uri = f"https://ipfs.io/ipfs/{ipfs_hash}?filename={filename}"
        print(image_uri)
        return image_uri
```

**Reply by cromewar** [reply]
I think the problem is this:
`pfs_url = "http://127.0.0.1:5001"`
If you are using base ipfs and want it to run into localhost you would have to run the node permanently in order to make [this](https://ipfs.io/ipfs/QmYEFoDo1ztBBXP83afPvovAHcoQjjbV5f3tcE1MQi5jXa?filename=0-WINDSWEPT.json) to work.
Have you consider use a external service such as pinata or moralis?
